### PR TITLE
agorakit: init at 1.9.2

### PR DESCRIPTION
**Pull request by Sh0g0-1758**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JulienMalka** [comment]
Added the module, largely inspired by the snipe-it module.

File: nixos/modules/services/web-apps/agorakit.nix [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+    #! ${pkgs.runtimeShell}
+    cd ${agorakit}
+    sudo() {
+      if [[ "$USER" != ${user} ]]; then
+        exec /run/wrappers/bin/sudo -u ${user} "$@"
+      else
+        exec "$@"
+      fi
+    }
+    sudo ${pkgs.php}/bin/php artisan "$@"
```
Comment (FliegendeWurst) [inline nixos/modules/services/web-apps/agorakit.nix:28]:
```suggestion
    sudo ${lib.getExe pkgs.php} artisan "$@"
```

File: nixos/modules/services/web-apps/agorakit.nix [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+      encryption = mkOption {
+        type = with types; nullOr (enum [ "tls" ]);
+        default = null;
+        description = "SMTP encryption mechanism to use.";
+      };
+    };
+
+    maxUploadSize = mkOption {
+      type = types.str;
+      default = "18M";
```
Comment (FliegendeWurst) [inline nixos/modules/services/web-apps/agorakit.nix:179]: Is this value the default upstream?

File: pkgs/by-name/ag/agorakit/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  '';
+
+  vendorHash = "sha256-5ypBA9Qb8jHzAtvNBHkJfsLIf3Pfw1LvYmHP/hED2ig=";
+  composerStrictValidation = false;
+
+  meta = with lib; {
+    description = "AgoraKit is web-based, open-source groupware for citizens' initiatives. By creating collaborative groups, people can discuss topics, organize events, store files and keep everyone updated as needed. AgoraKit is a forum, calendar, file manager and email notifier.";
+    homepage = "https://github.com/agorakit/agorakit";
+    license = licenses.agpl3Only;
+    platforms = platforms.all;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ag/agorakit/package.nix:36]:
```suggestion
  meta = {
    description = "Web-based, open-source groupware";
    longDescription = "AgoraKit is web-based, open-source groupware for citizens' initiatives. By creating collaborative groups, people can discuss topics, organize events, store files and keep everyone updated as needed. AgoraKit is a forum, calendar, file manager and email notifier.";
    homepage = "https://github.com/agorakit/agorakit";
    license = lib.licenses.agpl3Only;
    platforms = lib.platforms.all;
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/agorakit.nix (hunk 2 above)
Comment (fricklerhandwerk) [reply nixos/modules/services/web-apps/agorakit.nix:179]: There's no default upstream, this is a generic PHP setting

**Comment by infinisil** [comment]
This was merged after the branch-off, so this is not included in the current release by default, I created https://github.com/NixOS/nixpkgs/pull/359942 to move the release note entry to the correct location.

**Pull request merged**